pkg/dns/aws/dns.go

```diff
@@ -242,83 +241,102 @@ func (m *Provider) change(record *iov1.DNSRecord, zone configv1.DNSZone, action
 		return fmt.Errorf("unsupported record type %s", record.Spec.RecordType)
 	}
 	// TODO: handle >0 targets
-	domain, target := record.Spec.DNSName, record.Spec.Targets[0]
+	domain, targets := record.Spec.DNSName, record.Spec.Targets
+	updatedRecords := sets.NewString(strings.Join(m.updatedRecords.List(), ","))
```

Comment:
Why join the strings?  Don't forget to lock before access!  I think you want something like the following:
```go
	m.lock.Lock()
	updatedRecords := sets.NewString(m.updatedRecords.List()...)
	m.lock.Unlock()
```

Reply:
done